airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialReadUtil.java

```diff
@@ -420,16 +421,24 @@ static Optional<OrderedColumnInfo> getOrderedColumnInfo(final JdbcDatabase datab
     // if (stream.getStream().getSourceDefinedPrimaryKey().size() > 1) {
     // LOGGER.info("Composite primary key detected for {namespace, stream} : {}, {}",
     // stream.getStream().getNamespace(), stream.getStream().getName());
-    // } // TODO: validate the seleted column rather than primary key
-    final String clusterdIndexField = discoverClusteredIndexForStream(database, stream.getStream());
+    // }
+    final Map<String, List<String>> clusterdIndexField = discoverClusteredIndexForStream(database, stream.getStream());
+    final String streamName = getFullyQualifiedTableName(stream.getStream().getNamespace(), stream.getStream().getName());
+    final List<List<String>> primaryKey = stream.getStream().getSourceDefinedPrimaryKey();
     final String ocFieldName;
-    if (clusterdIndexField != null) {
-      ocFieldName = clusterdIndexField;
+
+    final List<String> clusterColumns = Optional.ofNullable(clusterdIndexField)
+        .map(map -> map.get(streamName))
+        .orElse(new ArrayList<>());
+
+    // Use the clustered index unless it is composite. Otherwise, default to the primary key.
+    if (clusterColumns.size() == 1) {
```

Comment:
What you added in `MssqlSource.getOCInfo()` you can put here as a static and use the same function for both testing and production.
If you can try moving it here so instead of copy you want to ideally test the production code.